### PR TITLE
Migrate styled components from css strings to objects. Use own faster implementation

### PR DESCRIPTION
**Pull request by terrysahaidak**
Fixes RNBW-2135

## What changed (plus any additional context for devs)

We are using styled-components a lot. But not just styled components, but also CSS strings.
All of the styles are represented as CSS strings like so:

```js
const Container = styled.View`
  width: 100px;
  height: 100px;
  background-color: ${({ theme, isDarkMode }) => isDarkMode ? theme.colors.light : theme.colors.dark};
`;
```

But there is a huge problem with that approach: React Native doesn't understand CSS strings. It can only use objects as styles. So in order to use CSS we need to parse it and transform it into objects. Some of these can be done during the app launch. But we can't "precompile" CSS strings where we use CSS inside of the string interpolation callback with props like this:`${props => css``}` . Because of the dynamic nature of this callback and we run it every time props change (sometimes we can't properly memorize it so it can be executed a lot of times which takes CPU time).
Also, styled-components creates "classNames" from each string using some hashing function to cache it. But basically it's still doing extra work here.

The solution is to use objects:

```js
const Container = styled.View({
  width: 100,
  height: 100,
  backgroundColor: ({ theme, isDarkMode }) => isDarkMode ? theme.colors.light : theme.colors.dark,
});
```

But turns out styled-components actually transform objects into the underlying array of CSS rules pairs and then use it to create classNames, hashes, etc - still lots of stuff we don't need at all.
So instead of using styled components we will use our own implementation which uses the same object-based API. So basically it's a drop-in replacement.

So now the code above will produce static styles and a callback to be executed on props change.

Basically now the app on android feels waaaay faster and smoother.

## Dev checklist for QA: what to test

Basically, it affects pretty much all the components of the app. So there can be some wrong styles because of some typos while migration.

Also, would be cool if someone could go through the files and check for any typos or wrong interpretation of the CSS styles into objects.

## Final checklist
[x] Assigned individual reviewers?
[x] Added labels?

### CODE REVIEW

File: src/components/animations/ButtonPressAnimation/NativeButton.js [hunk 1]
```diff
@@ -1,24 +1,30 @@
...
+    if (x !== 0.5) {
+      styles.left = `${x + 0.5 * (x > 0.5 ? 100 : -100)}%`;
+    }
+    if (y !== 0.5) {
+      styles.top = `${y + 0.5 * (y > 0.5 ? 100 : -100)}%`;
+    }
+
+    return styles;
+  }
+);
```
Comment (osdnk) [inline src/components/animations/ButtonPressAnimation/NativeButton.js:27]: Although I agree that the previous thing was not readable, I'd rather `return` inside `if`, not assign to the object. Sounds some "immutable:"

File: src/components/buttons/Button/Button.ios.js [hunk 2]
```diff
@@ -23,24 +23,35 @@ const ButtonShapeTypes = {
...
+  shadowColor: colors.alpha(
     isDarkMode ? colors.shadow : colors.blueGreyDark,
     isDarkMode && disabled ? 0.2 : 0.5
-  )};
-  shadow-offset: 0px 4px;
-  shadow-opacity: 0.2;
-  shadow-radius: 6;
-`;
+  ),
+  // TODO terry
```
Comment (osdnk) [inline src/components/buttons/Button/Button.ios.js:31]: TODO

File: src/components/buttons/rainbow-button/RainbowButtonBackground.js [hunk 3]
```diff
@@ -43,18 +43,20 @@ const RainbowButtonGradient = styled(RadialGradient).attrs(
     stops:
       type === RainbowButtonTypes.addCash ? [0, 0.544872, 1] : [0, 0.774321, 1],
   })
-)`
-  position: absolute;
-  transform: scaleY(0.7884615385);
-`;
+)({
+  position: 'absolute',
+  transform: [{ scaleY: 0.7884615385 }],
```
Comment (osdnk) [inline src/components/buttons/rainbow-button/RainbowButtonBackground.js:48]:
```suggestion
  transform: [{ scaleY: 0.7884615384 }],
```

I believe this looks significantly better, cc @christianbaroni

File: src/components/discover-sheet/DiscoverSearchContainer.js [hunk 4]
```diff
@@ -10,29 +10,29 @@ import React, {
...
   letterSpacing: 'roundedMedium',
   size: 'large',
   weight: 'semibold',
-}))`
-  ${ios ? '' : 'margin-top: -5;'}
-  margin-left: -3;
-  margin-right: 15;
-`;
+}))({
+  ...(ios ? {} : { marginTop: -5 }),
```
Comment (osdnk) [inline src/components/discover-sheet/DiscoverSearchContainer.js:32]:
```suggestion
  ...(android && { marginTop: -5 }),
```

File: src/components/expanded-state/chart/ChartExpandedStateHeader.js [hunk 5]
```diff
@@ -13,36 +11,18 @@ import {
...
-        onChange(
-          isScrubbing,
-          cond(isScrubbing, call([], enable), call([], disable))
-        ),
-      [disable, enable, isScrubbing]
-    )
-  );
-  return tabularNums;
-}
+})(({ showChart }) => ({
```
Comment (osdnk) [inline src/components/expanded-state/chart/ChartExpandedStateHeader.js:23]: @terrysahaidak those changes are unrel. We can merge this here unless you prefer to make this in the other PR

File: src/components/expanded-state/chart/chart-data-labels/ChartPercentChangeLabel.js [hunk 6]
```diff
@@ -63,7 +66,7 @@ export default function ChartPercentChangeLabel({
         );
       })();
 
-  const textProps = useAnimatedStyle(() => {
+  const textProps = useAnimatedProps(() => {
```
Comment (osdnk) [inline src/components/expanded-state/chart/chart-data-labels/ChartPercentChangeLabel.js:69]: it's ok, but please this PR is so big, let's do not do micro featuring here

File: src/components/expanded-state/custom-gas/FeesPanelTabs.js [hunk 7]
```diff
@@ -1,64 +1,64 @@
...
       ? isDarkMode
         ? colors.shadowBlack
         : color || colors.appleBlue
-      : colors.transparent};
-  shadow-offset: 0px 4px;
-  shadow-opacity: 0.3;
-  shadow-radius: 6;
-`;
+      : colors.transparent,
+  //  TODO terry
```
Comment (osdnk) [inline src/components/expanded-state/custom-gas/FeesPanelTabs.js:39]: TODO

File: src/components/expanded-state/unique-token/ZoomableWrapper.js [hunk 8]
```diff
@@ -35,29 +35,33 @@ const exitConfig = {
...
+  left: ({ containerWidth, width }) => -(width - containerWidth) / 2,
+  position: 'absolute',
+  top: -85,
+  width: ({ width }) => width,
+});
+
+const Container = styled(Animated.View)({
+  alignSelf: 'center',
+  shadowColor: ({ theme: { colors } }) => colors.shadowBlack,
+  // TODO terry
```
Comment (osdnk) [inline src/components/expanded-state/unique-token/ZoomableWrapper.js:49]: TODO

File: src/components/layout/InnerBorder.js [hunk 9]
```diff
@@ -1,15 +1,16 @@
...
-  opacity: ${({ opacity, theme: { isDarkMode } }) =>
-    isDarkMode ? 0 : opacity || 0.06};
-`;
+}).attrs({ pointerEvents: 'none' })(
+  ({ color, theme: { colors, isDarkMode }, opacity, width, radius }) => ({
+    ...position.coverAsObject,
+    borderColor: color || colors.black,
+    borderRadius: radius || 0,
+    borderWidth: width || 0.5,
+    opacity: isDarkMode ? 0 : opacity || 0.06,
```
Comment (osdnk) [inline src/components/layout/InnerBorder.js:12]:
```suggestion
    borderColor: color ?? colors.black,
    borderRadius: radius ?? 0,
    borderWidth: width ?? 0.5,
    opacity: isDarkMode ? 0 : opacity ?? 0.06,
```

File: src/components/layout/FlexItem.js [hunk 10]
```diff
@@ -1,11 +1,21 @@
...
-`;
+import styled from 'rainbowed-components';
+
+const FlexItem = styled.View(({ flex, grow, shrink }) => {
+  const props = {
+    flex:
+      isUndefined(flex) && isUndefined(grow) && isUndefined(shrink) ? 1 : flex,
+  };
+
+  if (typeof grow !== 'undefined') {
```
Comment (osdnk) [inline src/components/layout/FlexItem.js:10]: why not 
```suggestion
  if (grow !== undefined) {
```

File: src/components/fields/AddressField.js [hunk 11]
```diff
@@ -18,28 +18,28 @@ const AddressInput = styled(Input).attrs({
...
   spellCheck: false,
   weight: 'bold',
-})`
-  ${android && 'height: 56;'}
-  flex-grow: 1;
-  margin-top: 1;
-  z-index: 1;
-`;
+})({
+  ...(android ? { height: 56 } : {}),
```
Comment (osdnk) [inline src/components/fields/AddressField.js:22]: Here (and in many other places), I'd prefer:
```suggestion
  ...(android && { height: 56 }),
```

I will not post individual comments for this. What's the reason for preferring this over that?

File: src/components/gas/GasSpeedLabelPager.js [hunk 12]
```diff
@@ -1,40 +1,38 @@
...
   height: 30,
-})`
-  border: ${({ color, theme: { colors } }) =>
-    `2px solid ${color || colors.appleBlue}`};
-  ${padding(2.5, 4, android ? 2.5 : 3.5, 5)};
-  border-radius: 19;
-`;
+})({
+  ...padding.object(2.5, 4, android ? 2.5 : 3.5, 5),
+  borderCOlor: ({ color, theme: { colors } }) => color || colors.appleBlue,
```
Comment (osdnk) [inline src/components/gas/GasSpeedLabelPager.js:16]:
```suggestion
  borderCOlor: ({ color, theme: { colors } }) => color ?? colors.appleBlue,
```

File: src/components/modal/Modal.js [hunk 13]
```diff
@@ -1,34 +1,36 @@
...
-const Content = styled(Column).attrs({ shrink: 0 })`
-  border-radius: ${({ radius }) => radius};
-  height: ${({ height }) => height};
-  margin-top: ${({ fixedToTop }) => (fixedToTop ? 91 : 0)};
-  overflow: hidden;
-  padding-top: ${({ fullScreenOnAndroid }) =>
-    fullScreenOnAndroid && android ? getStatusBarHeight() : 0};
-  width: 100%;
-`;
+  // TODO terry
```
Comment (osdnk) [inline src/components/modal/Modal.js:19]: TODO

File: src/components/radio-list/RadioListItem.js [hunk 14]
```diff
@@ -1,19 +1,20 @@
...
   name: 'checkmarkCircled',
-}))`
-  box-shadow: 0px 4px 6px
-    ${({ theme: { colors, isDarkMode } }) =>
-      colors.alpha(isDarkMode ? colors.shadow : colors.appleBlue, 0.4)};
-  margin-bottom: 1px;
-  position: absolute;
-  right: 0;
-`;
+}))({
```
Comment (osdnk) [inline src/components/radio-list/RadioListItem.js:9]: TODO

File: src/components/settings-menu/BackupSection/WalletSelectionView.js [hunk 15]
```diff
@@ -18,70 +17,75 @@ import { ImgixImage } from '@rainbow-me/images';
...
 
 const GreenCheck = styled(CheckmarkIcon).attrs(({ theme: { colors } }) => ({
   color: colors.green,
-}))`
-  box-shadow: 0px 4px 6px
-    ${({ theme: { colors, isDarkMode } }) =>
-      colors.alpha(isDarkMode ? colors.shadow : colors.green, 0.4)};
-`;
+}))({
+  // TODO terry
```
Comment (osdnk) [inline src/components/settings-menu/BackupSection/WalletSelectionView.js:56]: TODO

File: src/components/settings-menu/SettingsSection.js [hunk 16]
```diff
@@ -53,47 +53,51 @@ export const SettingsExternalURLs = {
 
 const CheckmarkIcon = styled(Icon).attrs({
   name: 'checkmarkCircled',
-})`
-  box-shadow: 0px 4px 6px
-    ${({ theme: { colors, isDarkMode } }) =>
-      colors.alpha(isDarkMode ? colors.shadow : colors.blueGreyDark50, 0.4)};
-`;
+})({
+  // TODO terry
```
Comment (osdnk) [inline src/components/settings-menu/SettingsSection.js:57]: TODO

File: src/components/svg/SvgImage.js [hunk 17]
```diff
@@ -2,16 +2,17 @@
...
-const ImageTile = styled(ImgixImage)`
-  align-items: center;
-  justify-content: center;
-`;
+const ImageTile = styled(ImgixImage)({
+  alignItems: 'center',
+  justifyContent: 'center',
+});
 
+// TODO terry I messed up with some styles here
```
Comment (osdnk) [inline src/components/svg/SvgImage.js:15]: looks good

File: src/components/transaction/sections/TransactionConfirmationSection.js [hunk 18]
```diff
@@ -1,43 +1,44 @@
...
-  ${padding(android ? 1 : 7, 12, android ? 2 : 9, 11)};
-  border-radius: 40;
-  margin: auto;
-  margin-bottom: 17;
-  overflow: hidden;
-`;
+}))({
+  ...padding.object(android ? 1 : 7, 12, android ? 2 : 9, 11),
+  borderRadius: 40,
+  // TODO terry not sure about it
```
Comment (osdnk) [inline src/components/transaction/sections/TransactionConfirmationSection.js:26]: TODO

File: src/components/value-chart/Chart.js [hunk 19]
```diff
@@ -33,54 +33,54 @@ const ChartTimespans = [
...
-  shadow-radius: 4.5px;
-  width: 10px;
-`;
+const InnerDot = styled.View({
+  backgroundColor: ({ color }) => color,
+  borderRadius: 5,
+  height: 10,
+  shadowColor: ({ color, theme: { colors, isDarkMode } }) =>
+    isDarkMode ? colors.shadow : color,
+  // TODO terry
```
Comment (osdnk) [inline src/components/value-chart/Chart.js:61]: TODO

**Comment by osdnk** [comment]
As of now, I reviewed all 320 files and rewrites looks correct 👍

File: src/components/expanded-state/SupportedCountriesExpandedState.js [hunk 20]
```diff
@@ -58,7 +59,7 @@ const SupportCountriesExpandedState = () => {
         wiggleFactor={0}
       >
         <AssetPanel>
-          <Centered css={padding(19, 30, 24)} direction="column">
+          <Centered css={padding.object(19, 30, 24)} direction="column">
```
Comment (terrysahaidak) [inline src/components/expanded-state/SupportedCountriesExpandedState.js:62]: extract

**Review by terrysahaidak — COMMENTED** [review]
TODOs for me

File: src/components/expanded-state/SupportedCountriesExpandedState.js (hunk 20 above)
Comment (osdnk) [reply src/components/expanded-state/SupportedCountriesExpandedState.js]: don't you memo this anyway?

Comment (terrysahaidak) [reply src/components/expanded-state/SupportedCountriesExpandedState.js]: yes, it's memoized, but I was hoping for actually dripping deep memo at some point and instead making sure we can rely on shallow equal instead which is faster

File: src/components/3d/SimpleModelView.tsx [hunk 21]
```diff
@@ -7,9 +7,9 @@ import {
   ViewStyle,
 } from 'react-native';
 import { WebView } from 'react-native-webview';
-import styled from 'styled-components';
 import { ImgixImage } from '@rainbow-me/images';
 import { padding, position } from '@rainbow-me/styles';
+import styled from 'rainbowed-components';
```
Comment (markdalgleish) [inline src/components/3d/SimpleModelView.tsx:12]: Should this be under the `@rainbow-me` scope to make it a bit clearer that it's internal?

Also, if we do that, I wonder if it's better to just call it `@rainbow-me/styled-components` since it's trying to be a drop-in replacement (with missing features, obviously) and the scope makes it clear it's our fork of it.

File: src/components/ContactRowInfoButton.js [hunk 22]
```diff
@@ -18,44 +17,45 @@ import {
...
-}))`
-  height: 100%;
-  line-height: 29px;
-  width: 100%;
-  ${fontWithWidth(fonts.weight.bold)};
-`;
+}))({
+  height: '100%',
+  lineHeight: 29,
+  width: 100,
```
Comment (markdalgleish) [inline src/components/ContactRowInfoButton.js:56]: Is this meant to be `100%`?

File: src/components/add-cash/SupportButton.js [hunk 23]
```diff
@@ -6,14 +6,16 @@ import { Centered } from '../layout';
...
+
 const SupportButton = ({ label, onPress, ...props }) => {
   const { colors } = useTheme();
   return (
     <ButtonPressAnimation onPress={onPress} scaleTo={0.9}>
       <Centered
         backgroundColor={colors.alpha(colors.blueGreyDark, 0.06)}
         borderRadius={15}
-        css={padding(5, 10, 6)}
+        css={css}
```
Comment (markdalgleish) [inline src/components/add-cash/SupportButton.js:18]: Should this be a `style` prop?

File: src/components/asset-list/AssetListItemSkeleton.js [hunk 24]
```diff
@@ -39,36 +39,33 @@ const FakeRow = styled(Row).attrs({
   justify: 'space-between',
   paddingBottom: 5,
   paddingTop: 5,
-})(Row);
```
Comment (markdalgleish) [inline src/components/asset-list/AssetListItemSkeleton.js:42]: Why was `Row` passed here previously? Is that a mistake?

File: src/components/asset-list/RecyclerAssetList/index.tsx [hunk 25]
```diff
@@ -153,19 +153,21 @@ const isEqualDataProvider = new DataProvider((r1, r2) => {
...
-`;
-
-const StyledContainer = styled(View)`
-  display: flex;
-  flex: 1;
-  background-color: ${({ theme: { colors } }) => colors.white};
-  overflow: hidden;
-`;
+const StyledRecyclerListView = styled(RecyclerListView)({
+  // @ts-ignore
```
Comment (markdalgleish) [inline src/components/asset-list/RecyclerAssetList/index.tsx:157]: We should use `@ts-expect-error` so these get cleaned up if the types are fixed.

File: src/components/buttons/rainbow-button/RainbowButton.js [hunk 26]
```diff
@@ -47,28 +47,44 @@ const ButtonLabel = styled(Text).attrs(
...
+    width,
+    isDarkMode,
+    disabled,
+    strokeWidth,
+    theme: { colors },
+  }) => ({
+    ...shadow.buildAsObject(0, 5, 15, colors.shadow),
+    backgroundColor: colors.dark,
+    borderRadius: height / 2 + strokeWidth,
+    height: height,
```
Comment (markdalgleish) [inline src/components/buttons/rainbow-button/RainbowButton.js:64]:
```suggestion
    height,
```

File: src/components/buttons/rainbow-button/RainbowButtonBackground.js [hunk 27]
```diff
@@ -64,11 +66,11 @@ const InnerGradient = styled(RainbowButtonGradient).attrs(
...
   })
-)`
-  height: ${({ width }) => width};
-  top: ${({ height, width }) => -(width - height) / 2};
-  width: ${({ width }) => width};
-`;
+)(({ width, height }) => ({
+  height: width,
+  top: -(width - height) / 2,
+  width: width,
```
Comment (markdalgleish) [inline src/components/buttons/rainbow-button/RainbowButtonBackground.js:72]:
```suggestion
  width,
```

File: src/components/coin-icon/ChainBadge.js [hunk 28]
```diff
@@ -36,24 +36,24 @@ const sizeConfigs = {
...
-  z-index: 10;
-  elevation: 10;
-`;
+const IndicatorIconContainer = styled(Centered)(
+  ({ marginBottom, iconSize, badgeXPosition, badgeYPosition, position }) => ({
+    bottom: position === 'relative' ? 0 : badgeYPosition,
+    left: position === 'relative' ? 0 : badgeXPosition,
+    ...positions.sizeAsObject(iconSize),
+    elevation: 10,
+    marginBottom: marginBottom,
```
Comment (markdalgleish) [inline src/components/coin-icon/ChainBadge.js:51]:
```suggestion
    marginBottom,
```

File: src/components/discover-sheet/ListsSection.js [hunk 29]
```diff
@@ -46,26 +46,27 @@ const fetchTrendingAddresses = async coingeckoIds => {
...
+    ? {
+        backgroundColor: colors.alpha(colors.blueGreyDark, 0.06),
+        borderRadius: 12,
+        height: 30,
+        marginRight: 16,
+        paddingHorizontal: 8,
+        paddingTop: ios ? 6.5 : 4.5,
+      }
+    : {
+        marginRight: 16,
```
Comment (markdalgleish) [inline src/components/discover-sheet/ListsSection.js:60]: Looks like this was accidentally moved into the ternary and needs to be hoisted out.

File: src/components/discover-sheet/UniswapPoolsSection.js [hunk 30]
```diff
@@ -55,32 +55,32 @@ const ErrorMessage = ({ colors, children }) => (
...
-        padding-top: ${ios ? 6.5 : 4.5}px;
-      `}
-`;
+})(({ selected, theme: { colors }, titleColor }) =>
+  selected
+    ? {
+        backgroundColor: colors.alpha(titleColor, 0.06),
+        borderRadius: 12,
+        height: 30,
+        marginRight: 16,
```
Comment (markdalgleish) [inline src/components/discover-sheet/UniswapPoolsSection.js:70]: This property needs to be hoisted out of the ternary too.

File: src/components/discover-sheet/UniswapPoolsSection.js [hunk 31]
```diff
@@ -55,32 +55,32 @@ const ErrorMessage = ({ colors, children }) => (
...
+        paddingHorizontal: 8,
+        paddingTop: ios ? 6.5 : 4.5,
+      }
+    : {
+        marginRight: 16,
```
Comment (markdalgleish) [inline src/components/discover-sheet/UniswapPoolsSection.js:75]: This is duplicated across both states, so once it's been hoisted, this can be deleted.

File: src/components/expanded-state/ExpandedStateSection.js [hunk 32]
```diff
@@ -1,16 +1,16 @@
...
 const Container = styled(ColumnWithMargins).attrs({
   margin: 12,
-})`
-  ${padding(android ? 19 : 36, 19, 24)};
-  ${({ isNft }) => padding(android ? 19 : 36, isNft ? 24 : 19, 24)};
-  padding-top: ${({ isL2, isNft }) => (isL2 || isNft ? 24 : android ? 19 : 36)};
-`;
+})(({ isNft, isL2 }) => ({
+  ...padding.object(android ? 19 : 36, 19, 24),
+  paddingTop: isL2 || isNft ? 24 : android ? 19 : 36,
```
Comment (markdalgleish) [inline src/components/expanded-state/ExpandedStateSection.js:11]: This line was last in the string version, so looks like it needs to be moved down one line here to maintain the same precedence.

File: src/components/floating-panels/FloatingPanel.js [hunk 33]
```diff
@@ -12,18 +12,20 @@ export const FloatingPanelPadding = {
...
+
+const Container = styled(Column)(
+  ({ hideShadow, color, theme: { colors }, radius, minHeight, overflow }) => ({
+    ...(hideShadow ? {} : FloatingPanelShadow(colors)),
+
+    backgroundColor: color,
+    borderRadius: radius,
+    minHeight: minHeight || 0,
+    opacity: 1,
+    overflow: overflow,
```
Comment (markdalgleish) [inline src/components/floating-panels/FloatingPanel.js:25]:
```suggestion
    overflow,
```

File: src/components/investment-cards/PoolValue.js [hunk 34]
```diff
@@ -1,26 +1,26 @@
...
-const PoolValueWrapper = styled(Row)`
-  border-radius: ${({ simple }) => (simple ? 0 : 15)};
-  ${({ simple }) => (simple ? undefined : 'height: 30')};
-  ${({ simple }) => (simple ? undefined : padding(2, 9, 0))};
-`;
+const PoolValueWrapper = styled(Row)(({ simple }) => ({
+  borderRadius: simple ? 0 : 15,
+  ...(!simple && {
+    height: 30,
+    ...padding.object(2, 9, 9),
```
Comment (markdalgleish) [inline src/components/investment-cards/PoolValue.js:13]:
```suggestion
    ...padding.object(2, 9, 0),
```

File: src/components/sheet/sheet-action-buttons/SheetActionButtonRow.js [hunk 35]
```diff
@@ -1,23 +1,29 @@
...
-    )};
-  ${({ paddingBottom }) =>
-    paddingBottom ? `padding-bottom: ${paddingBottom};` : ``}
-  width: 100%;
-  z-index: 2;
-  elevation: -1;
-`;
+    ),
+
+    ...(paddingBottom ? { paddingBottom: paddingBottom } : {}),
```
Comment (markdalgleish) [inline src/components/sheet/sheet-action-buttons/SheetActionButtonRow.js:21]:
```suggestion
    ...(paddingBottom ? { paddingBottom } : {}),
```

File: src/screens/ExchangeModal.js [hunk 36]
```diff
@@ -68,19 +68,22 @@ const Wrapper = ios ? KeyboardFixedOpenLayout : Fragment;
...
-`;
-
-const Spacer = styled.View`
-  height: 20;
-`;
+})(
+  ios
+    ? {
+        ...position.sizeAsObject('100%'),
+        backgroundColor: ({ theme: { colors } }) => colors.transparent,
```
Comment (markdalgleish) [inline src/screens/ExchangeModal.js:75]: This property is duplicated across both states so should be hoisted out of the ternary.

File: src/screens/ExpandedAssetSheet.js [hunk 37]
```diff
@@ -33,11 +33,14 @@ const Container = styled(Centered).attrs({
...
-})`
-  ${position.cover};
-  ${({ deviceHeight, height }) =>
-    height ? `height: ${height + deviceHeight}` : null};
-`;
+})(({ deviceHeight, height }) =>
+  height
+    ? {
+        height: height + deviceHeight,
+        ...position.coverAsObject,
```
Comment (markdalgleish) [inline src/screens/ExpandedAssetSheet.js:40]: Duplicated, should be hoisted out of the ternary.

File: src/screens/SavingsSheet.js [hunk 38]
```diff
@@ -34,17 +33,20 @@ import { useNavigation } from '@rainbow-me/navigation';
...
-const Container = styled(Centered).attrs({ direction: 'column' })`
-  ${position.cover};
-  ${({ deviceHeight, height }) =>
-    height ? `height: ${height + deviceHeight}` : null};
-`;
+const Container = styled(Centered).attrs({
+  direction: 'column',
+})(({ deviceHeight, height }) =>
+  height
+    ? { height: height + deviceHeight, ...position.coverAsObject }
```
Comment (markdalgleish) [inline src/screens/SavingsSheet.js:47]: `position.coverAsObject` should be hoisted out of the ternary.

File: src/screens/SendConfirmationSheet.js [hunk 39]
```diff
@@ -43,56 +42,56 @@ import { useNavigation } from '@rainbow-me/navigation';
...
-  border-color: ${({ checked, theme: { colors } }) =>
-    colors.alpha(colors.blueGreyDark, checked ? 0 : 0.15)};
-  border-width: 2;
-  height: 20;
-  position: absolute;
-  width: 20;
-`;
+})(({ deviceHeight, height }) =>
+  height
+    ? { height: height + deviceHeight, ...position.coverAsObject }
```
Comment (markdalgleish) [inline src/screens/SendConfirmationSheet.js:51]: `position.coverAsObject` should be hoisted.

File: src/screens/SpeedUpAndCancelSheet.js [hunk 40]
```diff
@@ -53,38 +53,41 @@ const springConfig = {
...
-  height: 1000;
-  position: absolute;
-  bottom: -800;
-  width: 100%;
-`;
+})(({ deviceHeight, height }) =>
+  height
+    ? {
+        height: height + deviceHeight,
+        ...position.coverAsObject,
```
Comment (markdalgleish) [inline src/screens/SpeedUpAndCancelSheet.js]: `position.coverAsObject` should be hoisted.

File: src/styles/borders.js [hunk 41]
```diff
@@ -1,7 +1,7 @@
 import { upperFirst } from 'lodash';
-import { css } from 'styled-components';
 import colors from './colors';
 import position from './position';
+import { css } from 'rainbowed-components';
```
Comment (markdalgleish) [inline src/styles/borders.js]: Should `css` still be imported from `styled-components` since it's being passed a string?

File: src/styles/borders.js [hunk 42]
```diff
@@ -16,7 +16,7 @@ border.default = css`
 
 border.buildCircle = size => `
   ${position.size(size)};
-  border-radius: ${size / 2};
+  borderRadius: ${size / 2};
```
Comment (markdalgleish) [inline src/styles/borders.js]: This is still inside a string — should this be changed?

File: src/styles/buildFlexStyles.js [hunk 43]
```diff
@@ -1,5 +1,6 @@
-import { css } from 'styled-components';
+import { StyleSheet } from 'react-native';
 import position from './position';
+import { css } from 'rainbowed-components';
```
Comment (markdalgleish) [inline src/styles/buildFlexStyles.js]: Same question — should this be coming from `styled-components`?

File: src/styles/buildTextStyles.js [hunk 44]
```diff
@@ -61,7 +61,7 @@ const buildTextStyles = css`
   }}
 
   /* Font Size */
-  font-size: ${({ size = 'medium' }) =>
+  fontSize:  ${({ size = 'medium' }) =>
```
Comment (markdalgleish) [inline src/styles/buildTextStyles.js]: This is inside a string too. There's a few in this file.

File: src/styles/buildTextStyles.js [hunk 45]
```diff
@@ -99,4 +99,80 @@ const buildTextStyles = css`
...
+  opacity,
+  align,
+  tabularNums,
+  uppercase,
+}) => {
+  const styles = {
+    color: colors.get(color, theme.colors) || theme.colors.dark,
+  };
+
+  // function is used because of default argument values
```
Comment (markdalgleish) [inline src/styles/buildTextStyles.js]: Would it be better to provide default argument values in the `buildTextStyles.object` function itself?

File: src/styles/buildTextStyles.js [hunk 46]
```diff
@@ -99,4 +99,80 @@ const buildTextStyles = css`
...
+  (({ letterSpacing = 'rounded' }) => {
+    if (!isNil(letterSpacing)) {
+      styles.letterSpacing =
+        fonts.letterSpacing[letterSpacing] ?? letterSpacing;
+    }
+  })({ letterSpacing });
+
+  if (!(isNil(lineHeight) || (isEmoji && android))) {
```
Comment (markdalgleish) [inline src/styles/buildTextStyles.js]: I don't think this logic matches the original implementation which (naively inverted) is `!(isNil(lineHeight) || (isEmoji && android))`

File: tsconfig.json [hunk 47]
```diff
@@ -41,19 +42,17 @@
...
       "logger": ["src/utils/logger"],
       "react-native-cool-modals": ["src/react-native-cool-modals"],
-      "react-native-yet-another-bottom-sheet": ["src/react-native-yet-another-bottom-sheet"],
+      "react-native-yet-another-bottom-sheet": [
+        "src/react-native-yet-another-bottom-sheet"
+      ],
       "@rainbow-me/animated-charts": ["src/react-native-animated-charts/src"],
-      "react-native-shadow-stack": ["src/react-native-shadow-stack"]
+      "react-native-shadow-stack": ["src/react-native-shadow-stack"],
+      "@terrysahaidak/style-thing": ["src/styled-thing"],
```
Comment (markdalgleish) [inline tsconfig.json]: Should be deleted.

File: src/styles/borders.js (hunk 41 above)
Comment (terrysahaidak) [reply src/styles/borders.js]: It's re-exported from styled components. for backward compatibility with some existing code left what uses that css thing. It's something we want to remove in the future completely.

File: src/styles/borders.js (hunk 42 above)
Comment (terrysahaidak) [reply src/styles/borders.js]: There is object version too. Most of these have object version.

File: src/styles/buildTextStyles.js (hunk 44 above)
Comment (terrysahaidak) [reply src/styles/buildTextStyles.js]: There is object below.

File: src/components/asset-list/AssetListItemSkeleton.js (hunk 24 above)
Comment (terrysahaidak) [reply src/components/asset-list/AssetListItemSkeleton.js:42]: Yes, it was a mistake

File: src/components/asset-list/RecyclerAssetList/index.tsx (hunk 25 above)
Comment (terrysahaidak) [reply src/components/asset-list/RecyclerAssetList/index.tsx:157]: Yeah it makes sense, thanks

File: src/styles/borders.js (hunk 42 above)
Comment (markdalgleish) [reply src/styles/borders.js:19]: I mean that this property name uses camel case even though it's inside a string. Shouldn't this stay as `border-radius`?

File: src/styles/buildTextStyles.js (hunk 44 above)
Comment (markdalgleish) [reply src/styles/buildTextStyles.js:64]: Same as my other comment — I mean that this have been changed to camel case, but shouldn't it stay as `font-size` since it's in a string? If so, there's a few cases of this here.

**Review by markdalgleish — APPROVED** [review]
Just a couple of comments about the use of camel case property names within `css` strings.

Otherwise lgtm 👍

**Review by ibrahimtaveras00 — CHANGES_REQUESTED** [review]
Two issues posted in [RNBW-2171](https://linear.app/rainbow/issue/RNBW-2171/style-component-issues)

**Comment by skylarbarrera** [comment]
2 issues I ran into while looking into e2e, Change Wallet Sheet + Wallet connect requests are missing sheet border radius

**Comment by skylarbarrera** [comment]
resurfacing this smol issue:
AmountRow in TransactionConfirmationScreen : https://cloud.skylarbarrera.com/Screen-Recording-2022-01-10-14-37-50.mp4

**Comment by skylarbarrera** [comment]
Looks like there's some issues with coinIconFallback as well I believe 

Ex: https://linear.app/rainbow/issue/RNBW-2201/eth-icon-is-missing-on-android

File: src/utils/CoinIcons/CoinIcons.android.js [hunk 48]
```diff
@@ -1 +0,0 @@
-export default {};
```
Comment (osdnk) [inline src/utils/CoinIcons/CoinIcons.android.js:1]: please revert this or justify

**Review by osdnk — CHANGES_REQUESTED** [review]

**Review by osdnk — CHANGES_REQUESTED** [review]
Actually, please review if you included only intended files e.g. `src/rainbow-list`?

**Comment by terrysahaidak** [comment]
@ibrahimtaveras00 i just did yet another rebase so it should be good to check out. Thank you

File: src/rainbow-list/android/.cxx/abi_configuration_323y6l1b.json [hunk 49]
```diff
@@ -0,0 +1,14 @@
+{
```
[the remark this answers is not in the document]
Comment (terrysahaidak) [reply src/rainbow-list/android/.cxx/abi_configuration_323y6l1b.json]: Wow nice catch have no idea how it's got there

**Comment by osdnk** [comment]
@terrysahaidak please rebase again. PR by @benisgold broke linting

**Review by osdnk — APPROVED** [review]
I approve post CI

**Review by tchayen — CHANGES_REQUESTED** [review]
Just one thing.

File: src/components/value-chart/ExtremeLabels.js [hunk 50]
```diff
@@ -1,24 +1,24 @@
...
-  font-size: ${fonts.size.smedium};
-  font-weight: ${fonts.weight.bold};
-  letter-spacing: ${fonts.letterSpacing.roundedTighter};
-  position: absolute;
-  text-align: center;
-`;
+const Label = styled(Text)({
+  fontSize: fonts.size.smedium,
+  fontWeight: fonts.weight.bold,
+  letterspacing: fonts.letterSpacing.roundedTighter,
```
[the remark this answers is not in the document]
Comment (terrysahaidak) [reply src/components/value-chart/ExtremeLabels.js]: Thanks!

**Review by tchayen — APPROVED** [review]

**Review by ibrahimtaveras00 — APPROVED** [review]
Looks good on both OS's 👍🏽
